### PR TITLE
FIX: setdiff1d return type

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Following up on https://github.com/numpy/numpy/issues/5846#issuecomment-99466501.

### CODE REVIEW

**Comment by jaimefrio** [comment]
Wow, that's terse!

I'll need to look into the masked array behavior a little more carefully, probably tonight, but in principle it LGTM. Thanks Christian!

File: numpy/ma/tests/test_extras.py [hunk 1]
```diff
@@ -1109,6 +1109,8 @@ def test_setdiff1d(self):
         a = arange(10)
         b = arange(8)
         assert_equal(setdiff1d(a, b), array([8, 9]))
+        a = array([], np.uint32, mask=[])
+        assert setdiff1d(a, []).dtype is np.dtype('uint32')
```
Comment (jaimefrio) [inline numpy/ma/tests/test_extras.py]: You should use `assert_`, not `assert` here.

File: numpy/lib/tests/test_arraysetops.py [hunk 2]
```diff
@@ -286,6 +286,8 @@ def test_setdiff1d(self):
         assert_array_equal(c, ec)
 
         assert_array_equal([], setdiff1d([], []))
+        a = np.array((), np.uint32)
+        assert setdiff1d(a, []).dtype is np.dtype('uint32')
```
Comment (jaimefrio) [inline numpy/lib/tests/test_arraysetops.py]: Also `assert_`, not `assert`. You will need to add it to the functions imported from `numpy.testing` at the top of the file.

**Comment by jaimefrio** [comment]
Just two small testing nitpicks.

When you fix those, could you also squash everything into a single commit, start the commit message with `BUG:`, and add a `Fixes #5846` in the message body?

**Comment by christianbrodbeck** [comment]
I saw that I could use `assert_equal` too (rather than `assert_`) with a nicer error message, is that ok too?

**Comment by jaimefrio** [comment]
Now that I think about it, it is probably better: dtypes are not singletons, so that `is` comparisons was not the safest choice.

**Comment by jaimefrio** [comment]
In it goes. Thanks, Christian.

**Comment by christianbrodbeck** [comment]
Thanks!

**Pull request merged**